internal/pkg/csumio/csumreader.go

```diff
@@ -0,0 +1,76 @@
+package csumio
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+var (
+	ErrChecksumMismatch = errors.New("checksum mismatch")
+)
+
+type ChecksumReader struct {
+	dataFile             io.Reader
+	checksumFile         io.Reader
```

Comment:
The following comment is not reflected.

https://github.com/cybozu-go/fin/pull/162/files#r2517812587